packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart

```diff
@@ -0,0 +1,1176 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:io';
+
+// ignore_for_file: implementation_imports
+
+import 'package:bazel_worker/bazel_worker.dart';
+import 'package:bazel_worker/src/driver/driver.dart';
+import 'package:build/build.dart';
+import 'package:build_modules/build_modules.dart';
+import 'package:build_modules/src/module_library.dart';
+import 'package:build_web_compilers/builders.dart';
+import 'package:crypto/crypto.dart';
+import 'package:graphs/graphs.dart';
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:build_runner_core/build_runner_core.dart';
+import 'package:build_config/build_config.dart';
+import 'package:build_modules/builders.dart';
+import 'package:build_runner_core/src/generate/build_impl.dart';
+import 'package:build_runner_core/src/generate/options.dart';
+import 'package:path/path.dart' as path;
+import 'package:pool/pool.dart';
+import 'package:scratch_space/scratch_space.dart';
+import 'package:watcher/watcher.dart';
+import 'package:build_web_compilers/src/ddc_names.dart';
+import 'package:build_modules/src/errors.dart';
```

Comment:
We should definitely talk about these, we should just expose things if necessary so that we don't accidentally break you.

Reply:
Importing `build_impl.dart` makes me really nervous. Errors from `build_modules` is solvable - but importing from src in `build_runner_core` scares me.

Reply:
This is definitely not ideal, but the BuildImpl API was also trivial to integrate because it matches up exactly with how the flutter_tool currently handles invalidation.

The ideal API here would allow me to easily configurable all of the environment, but also expose all the hooks to handle build invalidation/running.

Reply:
RE: breaking changes. Because we pin versions I can handle rolling breakages myself, and we don't expect this code to be stable for quite some time

Reply:
OK, as long as you are happy with pinned versions for now that should be fine. These things do happen to evolve consistently though, just as an FYI.